### ISSUE TITLE
Meaning of sandbox_cgroup_only=false on cgroup memory accounting?

### ISSUE THREAD

**Issue by fengyehong** [kata-containers/kata-containers#4128]
When a kata container started, the related processes are:
- containerd-shim-kata-v2
- qemu
- virtiofsd

I understand that `containerd-shim-kata-v2` and `virtiofsd` should be placed under `/sys/fs/cgroup/memory/kata_overhead`. 
But qemu is just one process on the host, it should be under either `/sys/fs/cgroup/memory/kubepods` or `/sys/fs/cgroup/memory/kata_overhead`.

If qemu process under `/sys/fs/cgroup/memory/kubepods`, does it mean the VM overhead(eg. kernel memory) is added to pod?
If qemu process under `/sys/fs/cgroup/memory/kata_overhead`, does it mean all container workload memory usage not limited?

In my k8s node with sandbox_cgroup_only=false, i can ony observe kata_overhead memory usage:

```
# cat /sys/fs/cgroup/memory/kubepods/pod3a864680-ceaf-4acd-83fd-20a4d542b651/kata_94a6391dde707330a94acb3e684ee7048d085e7b6d9b1feda0cc1151b90606b2/memory.usage_in_bytes
0
# cat /sys/fs/cgroup/memory/kubepods/pod3a864680-ceaf-4acd-83fd-20a4d542b651/memory.usage_in_bytes
0
# cat /sys/fs/cgroup/memory/kata_overhead/94a6391dde707330a94acb3e684ee7048d085e7b6d9b1feda0cc1151b90606b2/memory.usage_in_bytes
1240653824
```

**Comment by liubin** [comment]
@fengyehong normally the overhead group will have no limitations, and sandbox cgroup will contain vCPU threads, you can check the `tasks` file.

You can read https://github.com/kata-containers/kata-containers/blob/main/docs/design/host-cgroups.md#sandbox_cgroup_only--false-default-setting for more info.

**Comment by fengyehong** [comment]
While cpu can be managed by 'threads', memory can only be managed by 'process'. So the qemu process can only be under either overhead cgroup or sandbox cgroup.

So my understanding is that: if qemu process under overhead cgroup(which is the case i observed), memory usage of the container is managed by qemu itself and cgroup config in VM, not by host machine. Am I right?

**Comment by liubin** [comment]
Yes, to get the correct Pod's stat, you should retrieve it from the guest.
